### DISCUSSION TITLE
Migrating to a new Linux (Debian) server

### DISCUSSION THREAD

**Discussion by tom76017** [GerryDazoo/Slinger#37] [Q&A]
I wanted to move my working instance of the server from my development environment to my user server.  It's running Let'sEncrypt and is a secure environment.  I'm assuming the one M1 can only service one sling.service.  The instance starts fine based on the log, but both VLC and the remote fail to connect either internally of externally.  Nothing is documented in the log.

This comment is from the install README.txt.  How does the "encrytion key" work?  i.e. only once, or everytime your restart the service.

> Note: The first time you run the server it will connect to the web to download a file with the proper encryption keys for you sling box. I'll try to intergrate that code as some point to remove this step. You python interpreter will need the "requests" module installed to make this work.

**Comment by GerryDazoo** [comment]
I’ll need to fix the readme. V3.07 no long needs to download the encryption keys. 

Once the server connects to the M1 you can stream as many simultaneous streams that your server/network can support.

I suspect you have a firewall issue that is blocking the connection requests from VLC etc. Turn off your firewall temporarily  to confirm.

 

*	Gerry

 

From: tom76017 ***@***.***> 
Sent: Monday, September 05, 2022 5:42 PM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] Migrating to a new Linux (Debian) server (Discussion #37)

 

I wanted to move my working instance of the server from my development environment to my user server. It's running Let'sEncrypt and is a secure environment. I'm assuming the one M1 can only service one sling.service. The instance starts fine based on the log, but both VLC and the remote fail to connect either internally of externally. Nothing is documented in the log.

This comment is from the install README.txt. How does the "encrytion key" work? i.e. only once, or everytime your restart the service.

Note: The first time you run the server it will connect to the web to download a file with the proper encryption keys for you sling box. I'll try to intergrate that code as some point to remove this step. You python interpreter will need the "requests" module installed to make this work.

**Reply by tom76017** [reply]
I'm still working on solving my VLC connection to my Debian server.   It still fails to connect.  I'm wondering if the problem is a python configuration issue.  I'm not a python programmer, but I saw this in a help file.   

**pip --trusted-host** 
I there something I need to do enable a **HTTPS** connection?

This is from VLC:
-- logger module started --
main: one instance mode ENABLED
main: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
access error: **HTTP** connection failure
http error: cannot connect to 192.168.11.50:8084
-- logger module stopped --

**Reply by GerryDazoo** [reply]
The software does not support an HTTPS streaming connection.
Make sure your firewall is allowing connection on port 8084. Assuming tht's the port you've configured in the [SERVER] section of your config.ini.
Next time please supply the output from the server startup and the complete URL you're giving to VLC.

**Reply by tom76017** [reply]
LAN Static IP's
Debian:    192.168.11.50
Slingbox:  192.168.11.70


**VLC ->  Sling.xspf**
############################################################################
			<location>http://192.168.11.50:8084/slingbox</location>
#############################################################################


**From Debian server ->  \tmp\sling.log**
#############################################################################
Version : 3.07 Running on Linux-5.10.0-18-amd64-x86_64-with-glibc2.31 pid= 1152
Using config file config.ini
Connection Manager Running on port 8084 with 10 max streams....
Streamer Running: starting up on  port 8084
 10 config.ini 8084
Checking for slingbox at ('192.168.11.70', 5301)
('192.168.11.70', 5301) OK
Using slingbox at  ('192.168.11.70', 5301)
Streamer:  Laundry Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
you should see some output on the server when it gets a connection. I'm 99% sure your firewall is blocking the requests.

**Reply by tom76017** [reply]
You are right.  Thanks for the help.  Works great!

- ufw allow 8084/tcp

Rule added
Rule added (v6)

**Comment by GerryDazoo** [comment]
Working

**Reply by tom76017** [reply]
Yes, you can close this.
